core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory.java

```diff
@@ -211,21 +236,30 @@ public void shutdown() {
      */
     private void updateBalancingState(ConnectivityState state, Status error) {
       List<Subchannel> activeList = filterNonFailingSubchannels(getSubchannels());
-      helper.updateBalancingState(state, new Picker(activeList, error, stickinessState));
+      ready = !activeList.isEmpty();
```

Comment:
The logic here is starting to get complicated, and it is no longer "obviously" correct.  Can you please document the params?